### PR TITLE
[kep] Add accepted KEP 2 on thread management

### PR DESCRIPTION
**Pull request by econchick**
Add the Accepted KEP 2 proposal. Its status will get updated to "Final" when the implementation is done.

@spotify/klio-devs please let me know if I've captured the relevant/important points in the Discussion section. Also, I think it might be a good idea to leave the final naming (e.g. arg names, class names, etc) and technical specifics to the eventual review of PR that implements this.

## Checklist for PR author(s)

- [ ] Format the pull request title like `[cli] Fixes bugs in 'klio job fake-cmd'`.
- [ ] Changes are covered by unit tests (no major decrease in code coverage %) and/or integration tests.
- [ ] Document any relevant additions/changes in the appropriate spot in `docs/src`.

### CODE REVIEW

File: docs/src/keps/kep-002.rst [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+Alternatives
+------------
+
+We could continue with the current situation and not manage the threads at all.
+
+We could also just provide the context manager and no decorator support, forcing the users to essentially opt-in and therefore understand/know what’s going on under the hood. We could also default to “off” or no thread concurrency management, again forcing the users to essentially opt-in.
+
+We could default the number of threads to 12, the `original default count <https://github.com/apache/beam/pull/10123/files#diff-07f4cace8fd5776c6add2121d4f58bfeb5256eb4ab6dfdd131e68348a967b8e3L193>`_ before Beam removed support for limiting worker threading [2]_.
+``12`` was considered too high for our current users, and would often set it to ``1`` or ``2``.
```
Comment (fallonchen) [inline docs/src/keps/kep-002.rst:321]: s/and would/who would

**Review by fallonchen — DISMISSED** [review]
:shipit:

File: docs/src/keps/kep-002.rst [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+Without the ability to configure exactly how many threads to use for a transform/pipeline, users will no longer be able to use Klio.
+
+
+References and Footnotes
+------------------------
+
+
+
+.. [1] The blocking point in the proposed :ref:`implementation <kep2-implementation>` is within the decorators before the decorated function/method is called, or at the point where the context manager starts. That is to say, the backpressure is within the scope of the transform being limited, not affecting other transforms.
+.. [2] `Patch <https://github.com/apache/beam/pull/10123>`_ where thread concurrency management is removed for Apache Beam v2.18.
```
[the remark this answers is not in the document]
Comment (econchick) [reply docs/src/keps/kep-002.rst:351]: me too! rST may be a bit of a pain, but that + sphinx is super powerful

File: docs/src/keps/kep-002.rst (hunk 1 above)
Comment (econchick) [reply docs/src/keps/kep-002.rst]: my human grammar / spell checker 🙃

**Pull request merged**